iroh-relay/src/dns/node_info.rs

```diff
@@ -137,17 +153,79 @@ impl NodeData {
     pub fn set_relay_url(&mut self, relay_url: Option<RelayUrl>) {
         self.relay_url = relay_url
     }
+
+    /// Sets the user data of the node data.
+    pub fn set_user_data(&mut self, user_data: Option<UserData>) {
+        self.user_data = user_data;
+    }
 }
 
 impl From<NodeAddr> for NodeData {
     fn from(node_addr: NodeAddr) -> Self {
         Self {
             relay_url: node_addr.relay_url,
             direct_addresses: node_addr.direct_addresses,
+            user_data: None,
+        }
+    }
+}
+
+// User defined data that can be published and resolved through node discovery.
+///
+/// Under the hood this is a UTF-8 String that is less than or equal to
+/// [`USER_DATA_MAX_LENGTH`] bytes.
+///
+/// Iroh does not keep track of or examine user defined data.
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct UserData(String);
+
+/// The max byte length allowed for user defined data.
```

Comment:
```suggestion
/// The max byte length allowed for user-defined data.
```

Reply:
Fixed, and also moved the const onto the `UserData` struct